src/ducks/apps/index.js

```diff
@@ -282,7 +290,73 @@ export function getFormattedInstalledApp(response, collectLink) {
 
 // only on the app initialisation
 export function initApp(lang) {
-  return fetchApps(lang)
+  return async dispatch => {
+    dispatch({ type: LOADING_APP })
+    await dispatch(initializeRealtime())
+    return dispatch(fetchApps(lang))
+  }
+}
+
+function listenAppUpdate(appResponse) {
+  return async (dispatch, getState) => {
+    if (appResponse.state === APP_STATE.ERRORED) {
+      const err = new Error('Error when installing the application')
+      dispatch({ type: INSTALL_APP_FAILURE, error: err })
+      throw err
+    }
+    if (appResponse.state === APP_STATE.READY) {
+      // FIXME: hack to handle node type from stack for the konnectors
+      const route =
+        appResponse.type === APP_TYPE.KONNECTOR || appResponse.type === 'node'
+          ? 'konnectors'
+          : 'apps'
+      const appFromStack = await cozy.client.fetchJSON(
+        'GET',
+        `/${route}/${appResponse.slug}`
+      )
+      // TODO throw error if collect is not installed
+      const collectApp = getState().apps.list.find(a => a.slug === 'collect')
+      const collectLink = collectApp && collectApp.related
+      return getFormattedInstalledApp(appFromStack, collectLink).then(app => {
+        // add the installed app to the state apps list
+        const apps = getState().apps.list.map(a => {
+          if (a.slug === app.slug) {
+            return Object.assign({}, a, app, { installed: true })
+          }
+          return a
+        })
+        return dispatch({ type: INSTALL_APP_SUCCESS, apps })
+      })
+    }
+  }
+}
+
+function initializeRealtime() {
+  return async dispatch => {
+    realtime
+      .subscribeAll(cozy.client, APPS_DOCTYPE)
+      .then(subscription => {
+        // HACK: the stack creates twice instead of updating
+        subscription.onCreate(app => dispatch(listenAppUpdate(app)))
+      })
+      .catch(error => {
+        console.warn &&
+          console.warn(`Cannot initialize realtime for apps: ${error.message}`)
+      })
+
+    realtime
+      .subscribeAll(cozy.client, KONNECTORS_DOCTYPE)
+      .then(subscription => {
+        // HACK: the stack creates twice instead of updating
```

Comment:
We should create an issue for this. Did you ?

Reply:
I don't know it it's an issue or a way of working about apps with the stack. Maybe @jinroh can say more about that?

Reply:
The only difficult thing for us to deactivate is the `create` event sent at the end of the installation, when the application is finally installed and committed into the database.

So simple solution for us is to simply remove the first "fake" `create` event. I think you do not really need it, because it is corresponding to a successful response to the HTTP request starting the installation process.

Reply:
I think the issue here is the fact of using `create` many times instead of using `update` event, isn't it?

Reply:
Yes, and as I was saying, it is awkward and "difficult" for make the second event an `update` event, since it actually correspond to the creation of the document in database.

Making the first event an `update` event does not really seem like a good idea either ? Or maybe it is ?

Reply:
Ok I understand. If we can't, I would keep things like that for now. Here we will just duplicate this code to use `onUpdate` as well, not more interesting than just the `onCreate`. If that concerns only the apps/konnectors, I would keep this comment to have the context every time we will read that code :)

Reply:
Thanks @jinroh for the explanations 👍